### PR TITLE
Global cache for JWKS

### PR DESCRIPTION
**Pull request by jhivandebenoit**
### Purpose

Reduce calls to JWKS endpoint  for Access token validation by caching all JWKS objects against their url. Before making a jwks call to a specific url we check the JWKS cache and try verify if it has the needed public key, otherwise we make the call.

### Automation tests
 - Unit tests added: No
 - Integration tests added: No
### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [ ] Assigned the project
- [x] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/AuthFilter.java [hunk 1]
```diff
@@ -54,6 +56,8 @@ public class AuthFilter implements Filter {
     private List<Authenticator> authenticators = new ArrayList<>();
     private static final Logger log = LogManager.getLogger(AuthFilter.class);
 
+    private static ConcurrentHashMap<String, JWKSet> jwksMap = new ConcurrentHashMap<>();
```
Comment (renuka-fernando) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/AuthFilter.java:59]: Any reason, jwksMap is initialized in AuthFilter and pass it into `JWTAuthenticator` -> `JWTValidator` without directly initialising it in JWTValidator? Is that for any future reference?

Comment (jhivandebenoit) [reply enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/AuthFilter.java:59]: no future reference in mind, its just how I thought of it. What you suggested would be better

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/validator/JWTValidator.java [hunk 2]
```diff
@@ -54,9 +55,9 @@ public class JWTValidator {
     private static final Logger logger = LogManager.getLogger(JWTValidator.class);
     private JWTTransformer jwtTransformer;
     private JWKSet jwkSet;
```
Comment (VirajSalaka) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/validator/JWTValidator.java:57]: We don't need JWKSet now ?

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/AuthFilter.java (hunk 1 above)
Comment (VirajSalaka) [reply enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/AuthFilter.java]: LGTM now, after moving the map to JWTValidator.

**Review by VirajSalaka — APPROVED** [review]

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/validator/JWTValidator.java (hunk 2 above)
Comment (VirajSalaka) [reply enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/validator/JWTValidator.java:57]: To minimize the perf impact, we decided to keep it as it is.

**Review by renuka-fernando — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : dev-deployment-v2 : 20230419.11

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20230419.11

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : dev-deployment-v2 : 20230419.11

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20230420.5

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : stage-deployment-v2 : 20230420.5

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20230420.5

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : stage-deployment-v2 : 20230420.5

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20230420.7

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20230420.7

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20230420.3

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : prod-deployment-v2 : 20230420.3

**Pull request merged**